core/lib/src/fs/static_file.rs

```diff
@@ -0,0 +1,494 @@
+        if raw_range_header.len() > 1 || raw_range_header.get(0).unwrap().find(",").is_some() {
+            // Multiple ranges are not supported.
+
+            // The spec allows for multiple ranges but we choose not to support them because the client may request
+            // very strange ranges (e.g. each byte separately, overlapping ranges, etc.) that could negatively
+            // impact the server. Ignore the header and serve the response normally.
+            return (false, None);
+        }
+
+        let range_header: Option<RangeHeaderValue> = req.headers().get_typed_headers().range();
+        if range_header.is_none() {
+            // Range header's value is invalid.
+            // Invalid
+            return (false, None);
+        }
+        let range_header = range_header.unwrap();
+
+        // Already verified above
+        assert_eq!(1, range_header.ranges.len());
+        let ranges = &range_header.ranges;
+        if ranges.is_empty() {
+            return (true, None);
+        }
+
+        if len == 0 {
+            return (true, None);
+        }
+
+        let range = ranges.first()
+            .map(|r| r.normalize(len))
+            .unwrap_or_default();
+
+        (range.is_some(), range)
+    }
+
+    fn apply_response_headers(&mut self, builder: &mut Builder<'_>, status: Status) {
+        builder.status(status);
+        if status.code < 400 {
+
+            // these headers are returned for 200, 206, and 304
+            // they are not returned for 412 and 416
+
+            if let Some(ct) = &self.content_type {
+                builder.header(ct.clone());
+            }
+            builder.raw_header(header_names::LAST_MODIFIED, self.last_modified.to_string());
+            builder.raw_header(header_names::ETAG, self.etag.to_string());
+            builder.raw_header(header_names::ACCEPT_RANGES, "bytes");
+            builder.raw_header(header_names::CONNECTION, "keep-alive");
+        }
+
+        if status == Status::Ok {
+            // this header is only returned here for 200
+            // it already set to the returned range for 206
+            // it is not returned for 304, 412, and 416
+            builder.raw_header(header_names::CONTENT_LENGTH, self.len.to_string());
+        }
+    }
+
+    fn send<'r>(self, req: &'r Request<'_>) -> response::Result<'static> {
+        let mut response = self.file.respond_to(req)?;
+        if let Some(ct) = self.content_type {
+            response.set_header(ct);
+        }
+        Ok(response)
+    }
+
+    fn send_range<'r>(mut self, _req: &'r Request<'_>) -> response::Result<'static> {
+        // do range
+        if let Some(ref range) = self.range {
+            let mut builder = Response::build();
+            let content_range_header = ContentRangeHeaderValue::from(range);
+            let from = content_range_header.from.unwrap();
+            let length = content_range_header.length.unwrap();
+            builder.header(content_range_header);
```

Comment:
Total length of the file is wrong. Below is response from server for file with size 79356416 bytes:

```
  content-range: bytes 29151232-79356415/50205184
```

According to https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Range after `/` there should be total file size, not remaining.

`wget` works after changing above lines to:
```rust
let mut builder = Response::build();
let from = range.from.unwrap();
let to = range.to.unwrap();
let content_range_header = ContentRangeHeaderValue::from((from, to, self.len));
let length = to - from + 1;
builder.header(content_range_header);
```

BTW. great PR, this is exactly what I need today :)

Reply:
Yes, you are right.